--- a/address.go
+++ b/address.go
@@ -137,12 +137,12 @@ func DecodeAddress(addr string, defaultNet *chaincfg.Params) (Address, error) {
 	// Bech32 encoded segwit addresses start with a human-readable part
 	// (hrp) followed by '1'. For Bitcoin mainnet the hrp is "bc", and for
 	// testnet it is "tb". If the address string has a prefix that matches
-	// one of the prefixes for the known networks, we try to decode it as
-	// a segwit address.
+	// the one defined in the defaultNet parameter, we try to decode it
+	// as a segwit address.
 	oneIndex := strings.LastIndexByte(addr, '1')
 	if oneIndex > 1 {
 		prefix := addr[:oneIndex+1]
-		if chaincfg.IsBech32SegwitPrefix(prefix) {
+		if strings.ToLower(prefix) == defaultNet.Bech32HRPSegwit+"1" {
 			witnessVer, witnessProg, err := decodeSegWitAddress(addr)
 			if err != nil {
 				return nil, err
